go/vt/vtgate/planbuilder/operators/ast_to_op.go

```diff
@@ -228,8 +228,8 @@ func getOperatorFromJoinTableExpr(ctx *plancontext.PlanningContext, tableExpr *s
 	switch tableExpr.Join {
 	case sqlparser.NormalJoinType:
 		return createInnerJoin(ctx, tableExpr, lhs, rhs)
-	case sqlparser.LeftJoinType, sqlparser.RightJoinType:
-		return createOuterJoin(tableExpr, lhs, rhs)
+	case sqlparser.LeftJoinType, sqlparser.RightJoinType, sqlparser.StraightJoinType:
+		return createLeftAndStraightJoin(tableExpr, lhs, rhs)
```

Comment:
if we use the same method of `left join` the inner join query for `straight join` will start disallowing `subquery` in the join condition